apps/studio/components/interfaces/Auth/Hooks/HooksListing.tsx

```diff
@@ -0,0 +1,191 @@
+      </Alert_Shadcn_>
+    )
+  }
+
+  // configuration for all auth hooks.
+  const hooks: Hook[] = HOOKS_DEFINITIONS.map((definition) => {
+    return {
+      ...definition,
+      enabled: authConfig?.[definition.enabledKey] || false,
+      method: extractMethod(
+        authConfig?.[definition.uriKey] || '',
+        authConfig?.[definition.secretsKey] || ''
+      ),
+    }
+  })
+
+  const nonEnterpriseHookOptions = hooks.filter((h) => !isValidHook(h) && !h.enterprise)
+  const enterpriseHookOptions = hooks.filter((h) => !isValidHook(h) && h.enterprise)
+
+  return (
+    <>
+      <FormHeader
+        title="Auth Hooks"
+        description="Use Postgres functions or HTTP endpoints to customize the behavior of Supabase Auth to meet your needs."
+        actions={
+          <DropdownMenu modal={false}>
+            <DropdownMenuTrigger asChild>
+              <Button
+                type="default"
+                iconRight={<ChevronDown className="w-4 h-4" strokeWidth={1} />}
+              >
+                Add hook
+              </Button>
+            </DropdownMenuTrigger>
+            <DropdownMenuContent className="w-76 p-0" align="end">
+              <div className="p-2">
+                {nonEnterpriseHookOptions.map((h) => (
+                  <DropdownMenuItem key={h.title} onClick={() => setSelectedHook(h.title)}>
+                    {h.title}
+                  </DropdownMenuItem>
+                ))}
+              </div>
+              {nonEnterpriseHookOptions.length > 0 && <DropdownMenuSeparator />}
+
+              <div className="bg-surface-200 p-2 -mt-2">
+                {subscription?.plan.id !== 'enterprise' && (
+                  <DropdownMenuLabel className="grid gap-1 bg-surface-200">
+                    <p className="text-foreground-light">Enterprise plan required</p>
+                    <p className="text-foreground-lighter text-xs">
+                      The following hooks are not available on{' '}
+                      <a
+                        target="_href"
+                        href="https://forms.supabase.com/enterprise"
+                        className="underline"
+                      >
+                        your plan
+                      </a>
+                      .
+                    </p>
+                  </DropdownMenuLabel>
+                )}
+                {enterpriseHookOptions.map((h) => (
+                  <DropdownMenuItem
+                    key={h.title}
+                    disabled={true}
+                    className="cursor-not-allowed"
+                    onClick={() => setSelectedHook(h.title)}
+                  >
+                    {h.title}
+                  </DropdownMenuItem>
+                ))}
+              </div>
+            </DropdownMenuContent>
+          </DropdownMenu>
+        }
+      />
+      <div className="first:rounded-t-md last:rounded-b-md pb-2">
```

Comment:
`first:rounded-t-md last:rounded-b-md` you need these on the cards themselves.
unless you did 

`[&>div]:first:rounded-t-md [&>div]:last:rounded-b-md` is probably what you wanted here.

i've added the comments to HookCard.tsx

Reply:
you will need `-space-y-px` on this div btw, so you don't get the double lines.
<img width="976" alt="Screenshot 2024-05-21 at 11 12 04 AM" src="https://github.com/supabase/supabase/assets/8291514/e3adbbfa-6da1-4bed-8f90-77e90e307717">

Reply:
Fixed.